### ISSUE TITLE
Add issues relations

### ISSUE THREAD

**Issue by sh2ka** [gogits/gogs#4571]
It would be nice to be able to link issues with each other like:
* `#234 is blocked by #204`
* `#234 resolves #125`
* etc.
Several links can be grouped like this:
```
Resolves
  #125 Some issue title
  #126 Another issue title
  ...
Blocked by
  #204 Issue 204 title
```

Corresponding issue should list corresponding links in it's own page like this:
```
Resolved by
  #234 Some major issue title
  ...
Blocks
  #235 Blocked issue title
  ...
```
So the used can see the lists of issue references and easily jump between them.

**Comment by Unknwon** [comment]
Hi, what makes this issue different from https://github.com/gogits/gogs/issues/1014?

**Comment by SystemDisc** [comment]
#1014 seems to just be a request to show when issues are referenced in other issues (in the comments), like as in the following image:
<img alt="" title="show when issues are referenced" src="https://i.imgur.com/GwxDrEN.png">
This is a severely needed feature.

However, this current issue seems to be a request for a feature to add specific relations, like marking one issue blocked by another issue or marking one issue as resolving another issue.

**Comment by Unknwon** [comment]
@SystemDisc thanks! Your explanation helped.

**Comment by SystemDisc** [comment]
I'm glad to be of service.

**Comment by BruinY** [comment]
+1
